compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -112,7 +112,13 @@ class PlainPrinter(_ctx: Context) extends Printer {
 
   /** String representation of a refinement */
   protected def toTextRefinement(rt: RefinedType): Closed =
-    (refinementNameString(rt) ~ toTextRHS(rt.refinedInfo)).close
+    val keyword = rt.refinedInfo match {
+      case _: TypeRef => "val "
+      case _: ExprType | _: MethodType => "def "
```

Comment:
there could also be PolyType, e.g. `def bar: Foo { def foo[T <: Any](t: T): String } = ???`

Reply:
for this case, `Foo` must be a class that already defines a polymorphic method